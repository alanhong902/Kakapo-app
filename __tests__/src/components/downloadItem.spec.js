import React from 'react';
import renderer from 'react-test-renderer';
import { map } from 'ramda';
import { newSoundObj } from 'utils/';
import { DownloadItem } from 'components/';

const soundProp = (props = {}) => {
  const obj = { ...newSoundObj, source: 'file', progress: 0.8, ...props };
  return map(x => (x === null ? 'wind' : x), obj);
};

test('<DownloadItem/> render', () => {
  const tree = renderer.create(<DownloadItem sound={soundProp()} />).toJSON();
  expect(tree).toMatchSnapshot();
});

test('<DownloadItem/> w/o image should render `no-image`', () => {
  const tree = renderer
    .create(
      <DownloadItem sound={soundProp({ source: 'youtubeStream', img: '' })} />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
