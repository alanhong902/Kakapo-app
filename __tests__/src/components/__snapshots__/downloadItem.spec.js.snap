// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DownloadItem/> render 1`] = `
<div
  className="download active"
>
  <div
    className="preview"
    style={
      Object {
        "backgroundImage": "url(http://data.kakapo.co/v2/images/light_wind.png)",
      }
    }
  />
  <span
    className="title"
  >
    wind
  </span>
  <div
    className="progress"
  >
    <div
      className="progress-barberpole"
      style={
        Object {
          "width": "80%",
        }
      }
    />
    <div
      className="progress-text"
    >
      80
      %
    </div>
  </div>
</div>
`;

exports[`<DownloadItem/> w/o image should render \`no-image\` 1`] = `
<div
  className="download active"
>
  <div
    className="no-image"
  />
  <span
    className="title"
  >
    wind
  </span>
  <div
    className="progress"
  >
    <div
      className="progress-barberpole"
      style={
        Object {
          "width": "80%",
        }
      }
    />
    <div
      className="progress-text"
    >
      80
      %
    </div>
  </div>
</div>
`;
